### PR TITLE
feat(Boards): AD-APARD32690-SL Board Support

### PR DESCRIPTION
**Pull request by BrentK-ADI**
### Description

Initial commit of the board support for the AD-APARD32690-SL board for the MAX32690.  Board files and updates to GPIO example project. [AD-APARD32690](https://wiki.analog.com/resources/eval/user-guides/ad-apard32690-sl)

### CODE REVIEW

**Comment by BrentK-ADI** [comment]
Commits so far include the actual board files and updates to the GPIO example, which looked to be the only example impacted by the different pin assignments.

Provide guidance on additional changes needed to complete the PR.

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Thanks @BrentK-ADI BSP looks good!  Below are some remaining items:

- [ ] Looks like the TFT demo also needs a little patching.  From the auto-tester:

```shell
MAX32690 (APARD): TFT_Demo failed.
Build command: make -r -j 8 TARGET=MAX32690 MAXIM_PATH=/home/runner/work/msdk/msdk BOARD=APARD FORCE_COLOR=1
Errors:
----------------------------------------
main.c:38:10: fatal error: tft_st7735.h: No such file or directory
   38 | #include "tft_st7735.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
make: *** 
[/home/runner/work/msdk/msdk/Libraries/CMSIS/Device/Maxim/GCC/gcc.mk:385: 
/home/runner/work/msdk/msdk/Examples/MAX32690/TFT_Demo/build/main.o] Error 1
make: *** Waiting for unfinished jobs....
----------------------------------------
```

FYI if you don't want to support a specific example (i.e. the board doesn't have a TFT), you can throw a `ERR_NOTSUPPORTED` from project.mk.  The build tester will log it as a known condition and pass it, and this will block mistaken users as well.

```Makefile
# project.mk
ifeq "$(BOARD)" "APARD"
$(error ERR_NOTSUPPORTED: This project is not supported for the APARD32690 board)
endif
```

- [ ] Add part under MAX32690 [support parts](https://github.com/analogdevicesinc/msdk/blob/main/USERGUIDE.md#supported-parts) in UG
- [ ] Add part to [BSP table](https://github.com/analogdevicesinc/msdk/blob/main/USERGUIDE.md#bsp-table) in UG
- [ ] clang-format (see [contributing](https://analogdevicesinc.github.io/msdk//CONTRIBUTING/#style-guide))

**Comment by BrentK-ADI** [comment]
@Jake-Carter Thanks for the tip on the ERR_NOTSUPPORTED.  This board doesn't have a display so might make sense to skip that example. 

I may have missed it, but didn't see any BSPs that include HyperRAM in them.  This board has 2 banks of HyperRAM attached to CS0 and CS1 respectively.  Should this be added similar to the External Flash, or leave it separate as just an example application?

Also, not for this commit, but future work, the board has a ADIN1110 MACPHY and MAXQ1065 Crypto Co-Processor.  At some point it'd be nice to tie these in, although we'll have to navigate the licensing issues as those SDKs fall under different licenses than the MSDK.

**Comment by sihyung-maxim** [comment]
> @Jake-Carter Thanks for the tip on the ERR_NOTSUPPORTED. This board doesn't have a display so might make sense to skip that example.
> 
> I may have missed it, but didn't see any BSPs that include HyperRAM in them. This board has 2 banks of HyperRAM attached to CS0 and CS1 respectively. Should this be added similar to the External Flash, or leave it separate as just an example application?
> 
> Also, not for this commit, but future work, the board has a ADIN1110 MACPHY and MAXQ1065 Crypto Co-Processor. At some point it'd be nice to tie these in, although we'll have to navigate the licensing issues as those SDKs fall under different licenses than the MSDK.

There exists an example using the MAX32690's HyperBus (HPB) Memory Controller peripheral with the HyperRAM.

https://github.com/analogdevicesinc/msdk/tree/main/Examples/MAX32690/HBMC

**Comment by BrentK-ADI** [comment]
The only examples that look to be problematic are anything based on I2C (EEPROM Emulator, I2C, Temp_Monitor).  The APARD board uses the alternate function I2C pins which are not automatically supported by the MSDK. See #949.  We can hold off on this PR until that is resolved, or come back to it later.

**Review by Jake-Carter — APPROVED** [review]
@BrentK-ADI looks good!  I opened #958 to fix those pin definitions.

> I may have missed it, but didn't see any BSPs that include HyperRAM in them. This board has 2 banks of HyperRAM attached to CS0 and CS1 respectively. Should this be added similar to the External Flash, or leave it separate as just an example application?

The more you can include in the BSP to match the on-board hardware the better.  I would suggest adding it similar to the external flash

**Comment by BrentK-ADI** [comment]
Rebased against the latest main which includes the I2C fixes.  Updated I2C related examples.  Needed to error out on two of them:

- I2C - Due to the errata and the use of the alternate pins on the board, 2 I2C buses arent available to perform the loopback example
- Temp_Monitor - No provisions on the board for 1.8V I2C

File: Libraries/Boards/MAX32690/APARD/examples.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Dual_Core_Sync/Dual_core_sync_arm
+Dual_Core_Sync/Dual_core_sync_riscv
+EEPROM_Emulator
+Flash
+Flash_CLI
+FreeRTOSDemo
+GPIO
+HBMC
+Hello_World
+I2C
```
Comment (Jake-Carter) [inline Libraries/Boards/MAX32690/APARD/examples.txt:44]: Remove unsupported examples from examples.txt as well (this is what Eclipse uses to populate its dropdown)

File: Libraries/Boards/MAX32690/APARD/Include/board.h [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+#ifdef LED_ON
+#undef LED_ON
+#endif
+#define LED_ON 1 /// Override active state of LEDs
+
+#ifndef EXT_FLASH_BAUD
+#define EXT_FLASH_BAUD 4000000
+#endif
+
+#define APARD /// Used in examples to control program flow
```
Comment (Jake-Carter) [inline Libraries/Boards/MAX32690/APARD/Include/board.h:73]: Just noticed this (minor quibble) - `BOARD_APARD` is defined on line 37.  Existing conventions prefers using `BOARD_xxx` for program flow

File: Libraries/Boards/MAX32690/APARD/examples.txt (hunk 1 above)
Comment (BrentK-ADI) [reply Libraries/Boards/MAX32690/APARD/examples.txt]: Removed I2C and Temp_Monitor in latest push

File: Libraries/Boards/MAX32690/APARD/Include/board.h (hunk 2 above)
Comment (BrentK-ADI) [reply Libraries/Boards/MAX32690/APARD/Include/board.h]: Updated as requested.

**Review by Jake-Carter — APPROVED** [review]

**Pull request merged**